src/utils/ExtrusionLine.cpp

```diff
@@ -27,4 +28,143 @@ void ExtrusionLine::appendJunctionsTo(LineJunctions& result) const
+        accumulated_area_removed += removed_area_next;
+
+        const coord_t length2 = vSize2(current - previous);
+        if (length2 < 25)
+        {
+            // We're allowed to always delete segments of less than 5 micron. The width in this case doesn't matter that
+            // much.
+            continue;
+        }
+
+        const coord_t area_removed_so_far = accumulated_area_removed + negative_area_closing; // close the shortcut area polygon
+        const coord_t base_length_2 = vSize2(next - previous);
+
+        if (base_length_2 == 0) //Two line segments form a line back and forth with no area.
+        {
+            continue; //Remove the vertex.
+        }
+        //We want to check if the height of the triangle formed by previous, current and next vertices is less than allowed_error_distance_squared.
+        //1/2 L = A           [actual area is half of the computed shoelace value] // Shoelace formula is .5*(...) , but we simplify the computation and take out the .5
+        //A = 1/2 * b * h     [triangle area formula]
+        //L = b * h           [apply above two and take out the 1/2]
+        //h = L / b           [divide by b]
+        //h^2 = (L / b)^2     [square it]
+        //h^2 = L^2 / b^2     [factor the divisor]
+        const coord_t height_2 = area_removed_so_far * area_removed_so_far / base_length_2;
+        if ((height_2 <= 1 //Almost exactly colinear (barring rounding errors).
+             && LinearAlg2D::getDistFromLine(current.p, previous.p, next.p) <= 1) // make sure that height_2 is not small because of cancellation of positive and negative areas
+             && static_cast<double>(llabs(current.w - next.w)) / nominal_wall_width <= maximum_width_change_ratio) // We shouldn't remove middle junctions of colinear segments if they change the width distinctively
+        {
+            continue;
+        }
+
+        if (length2 < smallest_line_segment_squared
+            && height_2 <= allowed_error_distance_squared) // removing the vertex doesn't introduce too much error.)
+        {
+            const coord_t next_length2 = vSize2(current - next);
+            if (next_length2 > smallest_line_segment_squared)
+            {
+                // Special case; The next line is long. If we were to remove this, it could happen that we get quite noticeable artifacts.
+                // We should instead move this point to a location where both edges are kept and then remove the previous point that we wanted to keep.
+                // By taking the intersection of these two lines, we get a point that preserves the direction (so it makes the corner a bit more pointy).
+                // We just need to be sure that the intersection point does not introduce an artifact itself.
+                Point intersection_point;
+                bool has_intersection = LinearAlg2D::lineLineIntersection(previous_previous.p, previous.p, current.p, next.p, intersection_point);
```

Comment:
Yup, that was also my thought on that. The point at index 1 will never actually be removed. But I thought it would be more confusing if I would start at index 2. Nevertheless, it will save one iteration for every toolpath, so I will start it at index 2 instead and I will explain why this happens in the function.

Reply:
Wait, come to think of it, point at index 1 can still be removed completely from the path if its distance from the previous is less than 5 microns. So we should check it.